### PR TITLE
Add database filter for the `assessment` workflow

### PR DESCRIPTION
**Pull request by william-conti**
## Changes
Added a database filter to allow the user to not crawl all the databases in Hive Metastore.

### Linked issues
Resolves https://github.com/databrickslabs/ucx/issues/937

### Functionality 

- [x] modified assessment

### Tests

- [X] manually tested
- [X] added unit tests
- [X] added integration tests
- [X] verified on staging environment (screenshot attached)

### CODE REVIEW

File: src/databricks/labs/ucx/hive_metastore/tables.scala [hunk 1]
```diff
@@ -72,20 +71,27 @@ def metadataForAllTables(databases: Seq[String], queue: ConcurrentLinkedQueue[Ta
...
-  val inventoryDatabase = configObj.get("inventory_database").toString()
-  return inventoryDatabase
+  return new Yaml().load(configContents).asInstanceOf[java.util.Map[String, Any]]
+}
+
+def getInventoryDatabase(configObj:java.util.Map[String, Any]): String ={
+  return configObj.get("inventory_database").toString()
+}
 
+def getDatabasesToFilter(configObj:java.util.Map[String, Any]): List[String] ={
```
Comment (nfx) [inline src/databricks/labs/ucx/hive_metastore/tables.scala:87]: Rename method to reflect configuration property name

File: tests/integration/hive_metastore/test_tables.py [hunk 2]
```diff
@@ -31,7 +31,10 @@ def test_describe_all_tables_in_databases(ws, sql_backend, inventory_schema, mak
         f"view={view.full_name}"
     )
 
-    tables = TablesCrawler(sql_backend, inventory_schema)
+    schema_c = make_schema(catalog_name="hive_metastore")
+    make_table(schema_name=schema_c.name)
+
+    tables = TablesCrawler(sql_backend, inventory_schema, [schema_a.name, schema_b.name])
```
Comment (nfx) [inline tests/integration/hive_metastore/test_tables.py:37]: And schema c?

File: tests/unit/hive_metastore/test_tables.py [hunk 3]
```diff
@@ -220,3 +220,13 @@ def test_is_supported_for_sync(table, supported):
...
+
+
+def test_tables_crawler_should_filter_by_database():
+    rows = {
+        "SHOW TABLES FROM hive_metastore.database": [("", "table1", ""), ("", "table2", "")],
+    }
+    backend = MockBackend(rows=rows)
+    tables_crawler = TablesCrawler(backend, "default", ["database"])
+    results = tables_crawler.snapshot()
+    assert len(results) == 2
```
Comment (nfx) [inline tests/unit/hive_metastore/test_tables.py:232]: Assert that there has been only that query

**Review by nfx — CHANGES_REQUESTED** [review]

File: tests/integration/hive_metastore/test_tables.py (hunk 2 above)
Comment (william-conti) [reply tests/integration/hive_metastore/test_tables.py:37]: I want to demonstrate that by applying a filter on database A and B, everything in database C won't be captured at all.

File: src/databricks/labs/ucx/config.py [hunk 4]
```diff
@@ -41,6 +41,9 @@ class WorkspaceConfig:  # pylint: disable=too-many-instance-attributes
     # Flag to see if terraform has been used for deploying certain entities
     is_terraform_used: bool = False
 
+    # Whether the assessment should capture a specific list of databases, if not specified, it will list all databases.
+    include_databases: list[str] | None = None
```
Comment (nfx) [inline src/databricks/labs/ucx/config.py:45]: can you document it in the readme?
https://github.com/databrickslabs/ucx?tab=readme-ov-file#open-remote-config-command

**Review by nfx — APPROVED** [review]

**Pull request merged**